From 06785b1fc6496976f53db3b77e5b66767d66e2e7 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Dec 2024 15:42:39 +0800
Subject: [PATCH 176/281] AOSCOS: platform: sd5075: convert to
 i2c_new_client_device() function

Follow commit 390fd0475af5 ("i2c: remove deprecated i2c_new_device API")
and revise i2c_new_device() to i2c_new_client_device(), there is no change
how the function is invoked.

Fixes: "AOSCOS: MIPS: Loongson 3: Add platform device drivers"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/platform/mips/sd5075.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/platform/mips/sd5075.c b/drivers/platform/mips/sd5075.c
index be42c5882290..02bb99821019 100644
--- a/drivers/platform/mips/sd5075.c
+++ b/drivers/platform/mips/sd5075.c
@@ -93,7 +93,7 @@ static int sd5075_probe(struct platform_device *dev)
 	info.addr = sdev->base_addr & ADDR_MASK;
 	info.platform_data = dev->dev.platform_data;
 	strncpy(info.type, "sd5075", I2C_NAME_SIZE);
-	sd5075_client = i2c_new_device(adapter, &info);
+	sd5075_client = i2c_new_client_device(adapter, &info);
 	if (sd5075_client == NULL) {
 		pr_err("failed to attach sd5075 sensor\n");
 		goto fail;
-- 
2.48.1

